python/metatensor-operations/metatensor/operations/to.py

```diff
@@ -116,19 +120,81 @@ def block_to(
                         " gradient tracking"
                     )
 
-    return _block_to(block, backend, dtype, device, requires_grad)
+    # Walk the tree of gradients without recursion
+    # (recursion is not supported by torchscript)
+    current_location: List[str] = []
+    current_block = block
+    transformed_blocks: List[TensorBlock] = []
+    last_visited = ""
+
+    while True:
+        print(len(transformed_blocks))
+        gradient_names = current_block.gradients_list()
+        n_gradients = len(gradient_names)
+        if last_visited == "":  # first time we see this block
```

Comment:
You should write a test for this code. It is quite complex and seems it will break easily.